graphql/src/execution/execution.rs

```diff
@@ -263,24 +271,39 @@ where
         }
     }
 
-    // Execute the root selection set against the root query type
-    if data_set.items.is_empty() {
-        // Only introspection
-        execute_selection_set(&ictx, &intro_set, introspection_query_type, initial_value)
```

Comment:
I just refactored the logic here so we only have the code to look up regular data once (since it's sorta more involved now) The introspection gets executed right after we got the regular data on lines 318-325